x/poolmanager/taker_fee.go

```diff
@@ -13,14 +13,21 @@ import (
 	txfeestypes "github.com/osmosis-labs/osmosis/v21/x/txfees/types"
 )
 
+func (k Keeper) GetDefaultTakerFee(ctx sdk.Context) sdk.Dec {
+	var defaultTakerFee sdk.Dec
+	k.paramSpace.Get(ctx, types.KeyDefaultTakerFee, &defaultTakerFee)
+	return defaultTakerFee
+}
+
 // SetDenomPairTakerFee sets the taker fee for the given trading pair.
 // If the taker fee for this denom pair matches the default taker fee, then
 // it is deleted from state.
 func (k Keeper) SetDenomPairTakerFee(ctx sdk.Context, denom0, denom1 string, takerFee osmomath.Dec) {
 	store := ctx.KVStore(k.storeKey)
 	// if given taker fee is equal to the default taker fee,
 	// delete whatever we have in current state to use default taker fee.
-	if takerFee.Equal(k.GetParams(ctx).TakerFeeParams.DefaultTakerFee) {
+	// TODO: This logic is actually wrong imo, where it can be valid to set an override over the default.
+	if takerFee.Equal(k.GetDefaultTakerFee(ctx)) {
```

Comment:
I don't think I understand the comment. It is valid to override over the default. Instead of storing every taker fee for every pair, we assume every pair is default, and if we pull the value for that pair and its non default, its an override.

Reply:
This code comment is on `SetDenomPairTakerFee`, for when its being set.

Its saying if you override a taker fee to a value that is default, delete the state entry for this, and make it go to default.

However then if the default changes, this one changes as well, whereas you really wanted it to always stay at the old default value.

(This is independent of the main point of this PR)